base/namedtuple.jl

```diff
@@ -129,6 +129,13 @@ function NamedTuple{names, T}(nt::NamedTuple) where {names, T <: Tuple}
     end
 end
 
+# Like NamedTuple{names, T} as a constructor, but omits the additional
+# `convert` call, when the types are known to match the fields
+struct _NamedTuple{names, T}; end
+@eval function _NamedTuple{names, T}(args::Tuple) where {names, T}
+    $(Expr(:splatnew, :(NamedTuple{names,T}), :(args)))
```

Comment:
Tt is not disallowed, but also not encouraged for a constructor to return a different type object. Should we make this a callable function instead, or simply inline this to the 2 places that require it?

Reply:
Yeah, fair enough. I'll just make it a generic function. I thought the sparams might be easier on inference, but I guess it doesn't really matter.